src/main/java/net/dv8tion/jda/core/requests/restaction/MessageAction.java

```diff
@@ -337,6 +340,7 @@ public MessageAction appendFormat(final String format, final Object... args)
 
     /**
      * Adds the provided {@link java.io.InputStream InputStream} as file data.
+     * <br><u>The stream will be closed upon execution!</u>
```

Comment:
When are given input streams actually closed? in the sending code?

Reply:
When they are read by the RequestBody it closes the stream.